### ISSUE TITLE
Driver shutting down error when using python to train the ao-cnn under gpu mode

### ISSUE THREAD

**Issue by wiw1234** [microsoft/O-CNN#92]
Hi,
I build the caffe-windows fellow the guide: https://github.com/BVLC/caffe/tree/windows.
In my setting, cuda version is 8.0, cudnn is 5.0. VS 2015 compiler, python 2.7， gpu 1080Ti.
It is ok to build the caffe and other octree tools. What's more, it is also ok to trian the ao-cnn under using the trian.sh.
But training the ao-cnn under python 2.7 using gpu, the error is below:
I0621 14:53:08.883052 29080 solver.cpp:56] Solver scaffolding done.
I0621 14:53:18.826083  7900 octree_database_layer.cpp:127] Prefetch batch: 56 ms.
F0621 14:53:20.640823 29080 syncedmem.hpp:39] Check failed: error == cudaSuccess (29 vs. 0)  driver shutting down
*** Check failure stack trace: ***

Training the ao-cnn under python 2.7 using cpu is fine! 

I have tried the newer version cuda 9.0 (or cuda 10.0), the issue still exist.

Any idea how I can fix this?
Thanks

**Comment by wiw1234** [comment]
Using python under GPU mode, training caffe network which don't contain octree-based  layer is no error.

**Comment by wang-ps** [comment]
Can you run the experiments in the command line mode (rather than using python)?

**Comment by wiw1234** [comment]
@wang-ps, much thanks for your reply.
Yes, there are no error when run the experiment like this：
D:/vvcc/o-cnn/caffe-windows/caffe-windows/scripts/build/tools/Release/caffe.exe train --solver=D:/vvcc/o-cnn/prototxt/full_32_32/image2shape.solver.prototxt

**Comment by wiw1234** [comment]
For detail, in my case,
Python code snippet is below:
solver_prototxt = 'D://vvcc//o-cnn//hao-cnn//prototxt//full_32_32//image2shape.solver.prototxt'
caffe.set_device(0)
caffe.set_mode_gpu()
solver = caffe.get_solver(solver_prototxt)
net = solver.net
net.forward()

The corresponding error is below:
......
I0622 22:10:14.636421 10268 net.cpp:242] This network produces output sum_depth_5_split_GT
I0622 22:10:14.636421 10268 net.cpp:242] This network produces output sum_depth_6_split_GT
I0622 22:10:14.636421 10268 net.cpp:242] This network produces output sum_depth_7_split_predict
I0622 22:10:14.636421 10268 net.cpp:242] This network produces output sum_imgFeatureData
I0622 22:10:14.637418 10268 net.cpp:255] Network initialization done.
I0622 22:10:14.637418 10268 solver.cpp:56] Solver scaffolding done.
F0622 22:10:14.944684 10268 syncedmem.hpp:39] Check failed: error == cudaSuccess (29 vs. 0)  driver shutting down
*** Check failure stack trace: ***



When python code snippet is like this:
solver_prototxt = 'D://vvcc//o-cnn/prototxt//full_32_32//image2shape.solver.prototxt'
caffe.set_device(0)
caffe.set_mode_gpu()
solver = caffe.get_solver(solver_prototxt)
solver.solve()


The corresponding is below:
.......
I0622 22:05:43.645001 14252 sgd_solver.cpp:105] Iteration 6, lr = 0.0001
I0622 22:05:43.741742 14252 solver.cpp:218] Iteration 7 (10.3646 iter/s, 0.0964827s/1 iters), loss = 6.08679
I0622 22:05:43.741742 14252 sgd_solver.cpp:105] Iteration 7, lr = 0.0001
I0622 22:05:43.828539 14252 solver.cpp:218] Iteration 8 (11.5535 iter/s, 0.0865537s/1 iters), loss = 6.68832
I0622 22:05:43.828539 14252 sgd_solver.cpp:105] Iteration 8, lr = 0.0001
I0622 22:05:43.956197 14252 solver.cpp:218] Iteration 9 (7.83682 iter/s, 0.127603s/1 iters), loss = 5.83919
I0622 22:05:43.957197 14252 sgd_solver.cpp:105] Iteration 9, lr = 0.0001
I0622 22:05:43.960157 14252 solver.cpp:447] Snapshotting to binary proto file D://vvcc//o-cnn//prototxt//model//_iter_10.caffemodel
I0622 22:05:44.138679 14252 sgd_solver.cpp:273] Snapshotting solver state to binary proto file D://vvcc//o-cnn//hao-cnn//prototxt//model//_iter_10.solverstate
I0622 22:05:44.298292 14252 solver.cpp:310] Iteration 10, loss = 6.25073
I0622 22:05:44.298292 14252 solver.cpp:315] Optimization Done.
F0622 22:05:44.524751 14252 syncedmem.hpp:39] Check failed: error == cudaSuccess (29 vs. 0)  driver shutting down

**Comment by wang-ps** [comment]
Thanks for pointing out this issue!
If you prefer python, please use the tensorflow-based implementation. Probably, I will not maintain the caffe-based implementation in the future. Sorry for the inconvenience.
